RSS Виртуальная клавиатура

Garza0 opened the pull request:
1. Задание - https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codejam-virtual-keyboard.md
2. Скриншот - https://imgur.com/a/c1N2BMT
3. Демо - https://garza0.github.io/virtual-keyboard/
07.04.2020 / 07.04.2020
Total ( 110 / 110)

**минимальный набор:**

- [x] реализована генерация DOM-элементов и body в index.html пустой изначально (может находится только тег script): +20
 - [x] нажатие на кнопку на физической клавиатуре подсвечивает кнопку на виртуальной (проверять следует нажатие цифр, букв, знаков препинания, backspace, del (если она присутствует), enter, shift, alt, ctrl, tab, caps lock, space, стрелки вниз-вверх-влево-вправо): +10

**стандартный набор:**

- [x]  есть переключение между русским и английским языком (сочетание клавиш для переключения языка должно быть указано на странице с клавиатурой), а также сохранение выбранного языка: +15
- [x]  клики мышкой по кнопкам на виртуальной клавиатуре или нажатие на кнопки физической клавиатуры, выводят символы в инпут (textarea): +15

**дополнительный набор:**

- [x]  реализована анимация нажатия на кнопку: +15

**технические требования:**

 - [x] использование в коде фишек стандарта ES6 и выше (classes, деструктуризация и т.д.): +15 

 - [x] использование eslint: +10

 - [x] требования к репозиторию, коммитам и PR выполнены: +10

**штрафы:**

 - [x] ошибки в консоли, связанные с исполняемым кодом (ошибки типа favicon.ico: Failed to load resource: the server responded with a status of 404 не учитываются) или предупреждения eslint-config-airbnb-base: -15

Garza0 committed:
feat: change repository

Я перезалил файлы в новый репозиторий,
так как сделал несколько ошибок вначале.
Старый репозиторий есть у меня в аккаунте,
называется 'codejam-virtual-keyboard'

Garza0 committed:
Merge branch 'master' into gh-pages

muraha commented on src/script.js:
```diff
@@ -0,0 +1,285 @@
+  language = sessionStorage.getItem('keyboard-language') ? sessionStorage.getItem('keyboard-language') : 'en';
+};
+
+restoreLangState();
```
restoration is better to do on window.onload event

Garza0 replied:
Fixed it. Thanks.

Garza0 committed:
feat: add window onload event

Garza0 committed:
Merge branch 'gh-pages' of https://github.com/Garza0/virtual-keyboard into gh-pages

muraha reviewed (approved):
All comments were left during live session in codebase directly. 
Git diff (patch) with changes and comments was provided to the student

Garza0 committed:
feat: make changes after live session with mentor

Garza0 committed:
feat: make changes after live session with mentor
-fix: error messages in browser console when action is on key which is not on the keys list
-better way to generate keyboard
-add keycode to the classlist, when generating keyboard
-css: add variables, access to the elements by classes
-better code style
-return in switch-case statement

Garza0 committed:
Merge branch 'gh-pages' of https://github.com/Garza0/virtual-keyboard into gh-pages

Garza0 commented:
> All comments were left during live session in codebase directly.
> Git diff (patch) with changes and comments was provided to the student

Сorrected all the places indicated by the @muraha . 
Fixed a bug when pressing a key not from the list of keys.

muraha commented on src/script.js:
```diff
@@ -149,34 +147,28 @@ function moveTextCursor(buttonKeyCode) {
       shiftState = true;
-      capsLkAndShiftHandler();
-      break;
+      return capsLkAndShiftHandler();
```
you do not need to return function.
just `return` instead of `break`

muraha commented on src/script.js:
```diff
@@ -149,34 +147,28 @@ function moveTextCursor(buttonKeyCode) {
     case 'shiftright':
       shiftState = true;
-      capsLkAndShiftHandler();
-      break;
```
```
capsLkAndShiftHandler();
      return
```

Garza0 replied:
fixed it

Garza0 committed:
fix: return in switch-case statement
